extension/beatsauthextension/authenticator.go

```diff
@@ -82,15 +82,11 @@ func (a *authenticator) configureTransport(transport *http.Transport) error {
 	if a.tlsConfig != nil {
 		// injecting verifyConnection here, keeping all other fields on TLSClientConfig intact
 
-		// should inject incoming root CA into our tls config
-		// so that we can build our custom verifyConnection method
-		// if root CA is empty, it uses system CA
+		// inject incoming root CA into our tls config
+		// because ca_trusted_fingerprint will be appended to incoming CA
 		a.tlsConfig.RootCAs = transport.TLSClientConfig.RootCAs
```

Comment:
Maybe I'm missing something, but here the *x509.CertPool is replace with `transport.TLSClientConfig.RootCAs` rather than appended or extended. Is this expected?